File: frontend/src/types/webnotenmanager.ts
export interface Webnotenmanager {
  id: string;
  schuleId: string;
  notenserverBaseUrl: string;
  oauthSecretEncrypted: string;
  createdAt: string;
  updatedAt: string;
}

export interface WebnotenmanagerRequest {
  notenserverBaseUrl: string;
  oauthSecretEncrypted: string;
}

export interface WebnotenmanagerWithSchuleData {
  id: string;
  schuleId: string;
  schulnummer?: number | null;
  schulname?: string | null;
  notenserverBaseUrl: string;
  oauthSecretEncrypted: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/src/main/java/de/schultraeger/domain/Webnotenmanager.java
package de.schultraeger.domain;

import java.time.Instant;
import java.util.UUID;

/**
 * Domain model for Webnotenmanager configuration.
 * Stores configuration for managing note servers and oauth credentials per school.
 */
public record Webnotenmanager(
        UUID id,
        UUID schuleId,
        String notenserverBaseUrl,
        String oauthSecretEncrypted,
        Instant createdAt,
        Instant updatedAt
) {
}
